Classes/Domain/Index/TcaIndexer/PagesIndexer.php

```diff
@@ -60,28 +67,73 @@ protected function prepareRecord(array &$record)
             }
         }
 
-        $record['content'] = $this->fetchContentForPage($record['uid']);
+        $record['media'] = $this->fetchMediaForPage($record['uid']);
+        $content = $this->fetchContentForPage($record['uid']);
+        if ($content !== []) {
+            $record['content'] = $content['content'];
+            $record['media'] = array_values(array_unique(array_merge($record['media'], $content['images'])));
+        }
         parent::prepareRecord($record);
     }
 
-    protected function fetchContentForPage(int $uid) : string
+    protected function fetchContentForPage(int $uid) : array
     {
-        $contentElements = $this->getQuery($this->contentTableService)->execute()->fetchAll();
+        if ($this->contentTableService instanceof TcaTableService) {
+            $contentElements = $this->contentTableService->getQuery()
+                ->execute()->fetchAll();
+        } else {
+            $contentElements = $GLOBALS['TYPO3_DB']->exec_SELECTgetRows(
+                $this->contentTableService->getFields(),
+                $this->contentTableService->getTableClause(),
+                $this->contentTableService->getWhereClause() .
+                sprintf(' AND %s.pid = %u', $this->contentTableService->getTableName(), $uid)
+            );
+        }
 
         if ($contentElements === null) {
             $this->logger->debug('No content for page ' . $uid);
-            return '';
+            return [];
         }
 
         $this->logger->debug('Fetched content for page ' . $uid);
+        $images = [];
         $content = [];
         foreach ($contentElements as $contentElement) {
+            $images = array_merge(
+                $images,
+                $this->getContentElementImages($contentElement['uid'])
+            );
             $content[] = $contentElement['bodytext'];
         }
 
-        // Remove Tags.
-        // Interpret escaped new lines and special chars.
-        // Trim, e.g. trailing or leading new lines.
-        return trim(stripcslashes(strip_tags(implode(' ', $content))));
+        return [
+            // Remove Tags.
+            // Interpret escaped new lines and special chars.
+            // Trim, e.g. trailing or leading new lines.
+            'content' => trim(stripcslashes(strip_tags(implode(' ', $content)))),
+            'images' => $images,
+        ];
+    }
+
+    protected function getContentElementImages(int $uidOfContentElement) : array
+    {
+        return $this->fetchSysFileReferenceUids($uidOfContentElement, 'tt_content', 'image');
```

Comment:
`image` is not the only field where FSC (Fluid Styled Content) can save them, right?

I think `TextMedia` uses the field `assets`.

Reply:
I'll keep it as it s and will change it once we need to.
Otherwise this will never end and we will fix existing issues all the time.